src/io/flutter/logging/FlutterLogView.java

```diff
@@ -0,0 +1,367 @@
+
+  }
+
+  @Override
+  public void addMessageFilter(@NotNull Filter filter) {
+
+  }
+
+  @Override
+  public void printHyperlink(@NotNull String hyperlinkText, @Nullable HyperlinkInfo info) {
+
+  }
+
+  @Override
+  public int getContentSize() {
+    return 0;
+  }
+
+  @Override
+  public boolean canPause() {
+    return false;
+  }
+
+  @NotNull
+  @Override
+  public AnAction[] createConsoleActions() {
+    // TODO(pq): consider actions (show up in a vertical bar).
+    return new AnAction[0];
+  }
+
+  @Override
+  public void allowHeavyFilters() {
+
+  }
+
+  @Override
+  public void dispose() {
+
+  }
+
+  @Nullable
+  @Override
+  public Object getData(String dataId) {
+    return null;
+  }
+
+  @Override
+  public JComponent getComponent() {
+    return toolWindowPanel;
+  }
+
+  @Override
+  public JComponent getPreferredFocusableComponent() {
+    return treeTable;
+  }
+
+  static class FlutterLogTreeTableModel extends DefaultTreeModel implements TreeTableModel {
+    @NotNull
+    private final Runnable updateRunnable;
+    @NotNull
+    private final FlutterLog log;
+    @NotNull
+    private final Alarm updateAlarm;
+
+    TreeTable treeTable;
+    private JTree tree;
+
+    public FlutterLogTreeTableModel(@NotNull FlutterLog log, @NotNull Disposable parent) {
+      super(new LogRootTreeNode());
+      this.log = log;
+
+      updateRunnable = () -> {
+        ((AbstractTableModel)treeTable.getModel()).fireTableDataChanged();
+        reload((TreeNode)getRoot());
+        treeTable.updateUI();
+      };
+
+      updateAlarm = new Alarm(Alarm.ThreadToUse.SWING_THREAD, parent);
+    }
+
+    private void update() {
+      if (!updateAlarm.isDisposed()) {
+        updateAlarm.cancelAllRequests();
+        updateAlarm.addRequest(updateRunnable, 0, ModalityState.stateForComponent(treeTable));
+      }
+    }
+
+    private LogTreeColumn getColumn(int index) {
+      return LogTreeColumn.values()[index];
+    }
+
+    @Override
+    public int getColumnCount() {
+      return LogTreeColumn.values().length;
+    }
+
+    @Override
+    public String getColumnName(int column) {
+      return getColumn(column).name;
+    }
+
+    @Override
+    public Class getColumnClass(int column) {
+      return getColumn(column).cls;
+    }
+
+    @Override
+    public Object getValueAt(Object node, int column) {
+      return getColumn(column).getValue(node);
+    }
+
+    @Override
+    public boolean isCellEditable(Object node, int column) {
+      return false;
+    }
+
+    @Override
+    public void setValueAt(Object aValue, Object node, int column) {
+    }
+
+    @Override
+    public void setTree(JTree tree) {
+      this.tree = tree;
+      treeTable = ((TreeTableTree)tree).getTreeTable();
+    }
+
+    // A temporary child counter; to be replaced w/ root.getChildCount()
+    int childCount;
+
+    public void onEvent(FlutterLogEntry entry) {
+      final MutableTreeNode root = (MutableTreeNode)getRoot();
```

Comment:
I think the event could come in on any thread; we should only modify the model here on the EDT thread. That may address the issues below.